tarogarden/caretaker.go

```diff
@@ -323,10 +323,25 @@ func (b *BatchCaretaker) seedlingsToAssetSprouts(ctx context.Context,
 		}
 
 		var groupKey *asset.GroupKey
+		// If the seedling has a group key specified and emission
+		// enabled, that group key was validated earlier. We need to
+		// sign the new genesis with that group key.
+		if seedling.HasGroupKey() {
+			groupKey, err = asset.DeriveGroupKey(
+				b.cfg.GenSigner,
+				seedling.GroupInfo.GroupKey.RawKey,
+				*seedling.GroupInfo.Genesis, &assetGen,
+			)
+			if err != nil {
+				return nil, fmt.Errorf("unable to tweak	group "+
+					"key: %v", err)
+			}
+		}
+
 		// If emission is enabled, then we'll need to generate another
 		// public key, then use that to derive the key group signature
 		// along with the tweaked key group.
-		if seedling.EnableEmission {
+		if seedling.EnableEmission && !seedling.HasGroupKey() {
```

Comment:
Could put this under a switch statement with the above clause. Then you wouldn't need this extra clause here.